Aplib.Core/Intent/Actions/GuardedAction.cs

```diff
@@ -1,52 +1,55 @@
-﻿using System;
+﻿using Aplib.Core.Belief;
+using Aplib.Core.Intent.Actions;
 
 namespace Aplib.Core.Intent.Actions
 {
     /// <summary>
     /// Describes an action that can be executed and guarded with a query that stores the result of the guard.
     /// The result can be used in the effect.
     /// </summary>
+    /// <typeparam name="TBeliefSet">The belief set of the agent.</typeparam>
     /// <typeparam name="TQuery">The type of the query of the action</typeparam>
-    public class GuardedAction<TQuery> : Action
+    public class GuardedAction<TBeliefSet, TQuery> : Action<TBeliefSet>
+        where TBeliefSet : IBeliefSet
     {
         /// <summary>
         /// Gets or sets the result of the guard.
         /// </summary>
-        protected TQuery? StoredGuardResult { get; set; }
+        protected TQuery? _storedGuardResult { get; set; }
 
         /// <summary>
         /// Gets or sets the effect of the action.
         /// </summary>
-        protected new System.Action<TQuery> Effect { get; set; }
+        protected new System.Action<TBeliefSet, TQuery> _effect { get; set; }
 
         /// <summary>
         /// Gets or sets the guard of the action.
         /// </summary>
-        protected new Func<TQuery?> Guard { get; set; }
+        protected new System.Func<TBeliefSet, TQuery?> _guard { get; set; }
```

Comment:
System is hier overbodig, aangezien we zelf geen `Func` aanbieden.

```suggestion
        protected new Func<TBeliefSet, TQuery?> _guard { get; set; }
```

Reply:
Kwam deze niet omdat we geen using system kunnen gebruiken hier?

Reply:
Klopt

Reply:
Waarom zou dat niet kunnen?